src/ae_kqueue.c

```diff
@@ -132,31 +132,27 @@ static int aeApiPoll(aeEventLoop *eventLoop, struct timeval *tvp) {
     aeApiState *state = eventLoop->apidata;
     int retval, numevents = 0;
 
+    struct timespec timeout = {0, 0};
     if (tvp != NULL) {
-        struct timespec timeout;
         timeout.tv_sec = tvp->tv_sec;
         timeout.tv_nsec = tvp->tv_usec * 1000;
-        retval = kevent(state->kqfd, NULL, 0, state->events, eventLoop->setsize,
-                        &timeout);
-    } else {
-        retval = kevent(state->kqfd, NULL, 0, state->events, eventLoop->setsize,
-                        NULL);
     }
+    retval = kevent(state->kqfd, NULL, 0, state->events, eventLoop->setsize, tvp ? &timeout : NULL);
```

Comment:
Usually we don't want to accept refactoring changes that don't add any new value, since it is extra work to review it and it affects the git history of the file and the changed lines.

Maybe in this case it is worth it since it removes some duplicated code (the call to kevent) but I think it is low prio, so @panjf2000 I think you should not be disapointed if it is not merged.

Reply:
Thank you for your reply! I see your point, and this is just a cleanup, also don't worry about it, I can live with whatever the decision is.

Reply:
when reading the description (using the same pattern as in other "ae" backends) and discussion, i initially thought that i'm gonna choose to take it because if we have several blocks with a different pattern, then it's easier and less bug prone to update one and replicate the change to the others.

but that was before i looked at the diff and saw that it's really local (3 adjacent lines, rather than some more complicated mechanism, or a long if-else chain).
also, considering the current code isn't too complicated to follow (e.g. causing it to be bug-prone for being hard to read and maintain), i don't see a reason to change it.

sorry.

i'll take this opportunity to note that the lifecycle of a change in the code doesn't end when the PR is merged, it can echo for quite a long time, being propagated to branches and forks, be reviewed again when new bugs are investigated, and so on.. so despite wasting more words on this comment than what it would have taken me to merge it, i may still save myself and others time by not merging it.

p.s. we do have cases from time to time there's a really innocent change (sometimes backported to older releases as an optimization), and we later find the change (despite looking trivial) was hiding a bug.